### PR TITLE
Feature: tagging rules reset

### LINKED ISSUE #7520: A way to remove all tagging rules
I have exported my tagging rules of which there are many and cleaned them up.

Im happy with the import process and re-running tag all from the console.

However I can find no way to remove all rules so that I only get what ive imported and not all the old ones.

If you have many rules clicking delete and having the page refresh after a wait is time consuming and there should be an easier way.

### PR DESCRIPTION
**Pull request by KevinSJ**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | yes
| CHANGELOG.md  | no
| License       | MIT

Implements and closes #7520 

I put the reset tagging rules button in the reset area.

The button will only show up if the current user has tagging rules.

My first PR using PHP :smile:.Happy to make any changes.

### CODE REVIEW

File: src/Repository/TaggingRuleRepository.php [hunk 1]
```diff
@@ -12,4 +12,18 @@ public function __construct(ManagerRegistry $registry)
...
         parent::__construct($registry, TaggingRule::class);
     }
+
+    /**
+     * Remove all tagging rule for a config.
+     * Used when a user wants to reset.
+     *
+     * @param int $configId
+     */
+    public function removeAllTaggingRules($configId)
```
Comment (j0k3r) [inline src/Repository/TaggingRuleRepository.php:22]:
```suggestion
    public function removeAllByConfigId($configId)
```

**Review by j0k3r — COMMENTED** [review]
Good job for your first PHP PR :v:
Just few typos to fix, otherwise, it's good!

**Comment by KevinSJ** [comment]
Thank you, have updated. Please review when you have time.

**Review by nicosomb — APPROVED** [review]

**Comment by KevinSJ** [comment]
Thanks, will find time to squash commits tomorrow.

**Comment by KevinSJ** [comment]
@j0k3r I have squashed the commits as requested. Thanks!

**Review by j0k3r — APPROVED** [review]

**Pull request merged**